src/Relude/Extra/Type.hs

```diff
@@ -114,3 +117,17 @@ Maybe (Snd (Int, Text)) :: *
 type family Snd (t :: k) :: k' where
     Snd '(_, y) = y
     Snd  (_, y) = y
+
+
+{- | Check that a type is an element of a list:
+>>> :kind! (Elem Bool '[Int, Bool])
+(Elem Bool '[Int, Bool]) :: Bool
+= 'True
+>>> :kind! (Elem String '[Int, Bool])
+(Elem String '[Int, Bool]) :: Bool
+= 'False
+-}
+type family Elem (e :: t) (es :: [t]) :: Bool where
+    Elem _ '[]       = 'False
+    Elem x (x ': xs) = 'True
+    Elem x (y ': xs) = Elem x xs
```

Comment:
I wonder, will it work if you write like this? :thinking: 
```suggestion
    Elem x (_ ': xs) = Elem x xs
```

Reply:
Yes, it works! 🎉